x-pack/plugins/canvas/public/lib/aeroelastic/select.ts

```diff
@@ -6,9 +6,11 @@
 
 import { ActionId, Json, PlainFun, Selector, State } from '.';
 
-export const select = (fun: PlainFun): Selector => (...fns) => {
+type SelectFn = (state: State) => Json;
```

Comment:
There's a type for it in `index.d.ts` (`Resolve`, as yet unexported)

This would also have future use when typing eg. `layout.js` assuming we don't convert it to some simpler pipeline expression first, so `index.d.ts` might still be a good place - also, if we eventually extract the layout functionality, it's nice to have one place to go for the types esp. those that are used outside the module

Reply:
Sure, I'll move it to the `d.ts` file.